src/sqlancer/postgres/gen/PostgresTableGenerator.java

```diff
@@ -185,6 +189,14 @@ private void generatePartitionBy() {
         sb.append(")");
     }
 
+    private void generateUsing() {
+        if (Randomly.getBoolean() || isPartitionedTable) {
```

Comment:
I would swap the checks around. When we know its a partitioned table we don't need to pull a random number from the rng.

I would actually extract it in its own `if` statement and leave a comment on why we are not generating a USING clause when the table is partitions.

Sometime like
```java
// Partitioned tables don't have any storage associated with the table.
// Hence the USING syntax is not valid syntax for partitioned tables.
```

Even though you can do this on this if statement as well, the code is general slightly easier to follow by a reader when such checks are made separately. The partition check here is very specific to the `USING` clause, where the Random check is all over the place to make every run behave differently if seeded with a different seed.